### PR TITLE
Normalized spacing/brackets in pseudocode

### PR DESCRIPTION
**Pull request by pjotrek-b**
In order to make the pseudocode less susceptible to misinterpretation, I've added {} even for one-liners for all "if" and "for".

For the sake of consistency, I also normalized the spacing of "if", "for" and "while" calls as follows:
  * Space between command (if, for, while) and arguments
  * Space between operators.
  * No space after/before closing (). Unless inside calls a function, to avoid "))" for better readability.

Although these changes are mostly cosmetic, I think it's good to have consistent and well-readable code syntax throughout a document.

### CODE REVIEW

**Comment by pjotrek-b** [comment]
I've also added additional typo corrections.

File: ffv1.md [hunk 1]
```diff
@@ -704,17 +708,20 @@ log2_run[41]={                                                |
 };                                                            |
                                                               |
 if (run_count == 0 && run_mode == 1) {                        |
-    if (get_bits(1)) {                                        |
+    if ( get_bits(1) ) {                                      |
```
Comment (JeromeMartinez) [inline ffv1.md]: spaces near parenthesis are added here as you indicated ("Space between command (if, for, while) and arguments") but I don't see the change for other parts ("if (x + run_count <= w)", "if (run_index)"... below and lot of others), or did I miss something?

Comment (pjotrek-b) [reply ffv1.md]: It seems inconsistent, but I followed a rule:
  * If the statement in brackets ends with a bracket, I've kept/added the space, so it's clearer to read.
  * If it does not end in brackets, there's no need for space for readability.

Comment (JeromeMartinez) [reply ffv1.md]: Ho, sorry, right you wrote it.
On my side I think it looks weird, I would prefer one of them only (all with space or none with space), but nothing big and we can keep that as is. @michaelni any preference?

Comment (michaelni) [reply ffv1.md]: the "mixed" style looks a bit odd to me too. Not that it really matters, its just spaces ...

Comment (pjotrek-b) [reply ffv1.md]: I actually took the "))"-avoiding from existing code-styles in the paper. Thought it was on purpose that it was already mixed therefore.

Anyways: I can normalize all to non-space. No problem.

Shall I?

Comment (pjotrek-b) [reply ffv1.md]: Okay, I've now checked and seen that it's actually just this single if() with extra spaces. Thought there were more cases like that... :see_no_evil:

So I've now changed this so everything should be conform now :smile:

Comment (JeromeMartinez) [reply ffv1.md]: I may have missed something but I see that you removed space from get_ur_golomb() part but not the one I am commenting (below if (run_count == 0 && run_mode == 1)).
PS: small stuff, just spaces, but as I find nothing else... :).

**Comment by michaelni** [comment]
the spaces within while() seems inconsistent

**Comment by pjotrek-b** [comment]
I applied the same "no ))" rule to while.
I've now also removed the spaces there :smile:

btw: I've changed `!byte_aligned()`to `!byte_aligned( )`, since all empty parentheses in the rest of the document have a space between.

I hope I didn't miss anything anymore?

**Comment by pjotrek-b** [comment]
OMG! :scream:
Now I know why I couldn't find " ( " anymore!
It's not spaces (0x20) in the document, but something else (0xC2A0)...

ffv1_specs_space-0xc2a0

**Comment by pjotrek-b** [comment]
I've never seen this before, and I'll check if this was done by some process on my side...

**Comment by JeromeMartinez** [comment]
> It's not spaces (0x20) in the document, but something else (0xC2A0)...

it is the UTF-8 code for [non-breaking space](https://en.wikipedia.org/wiki/Non-breaking_space).
Not bad in the idea, more complex in the writing, maybe could be replaced by classic spaces.

**Comment by pjotrek-b** [comment]
Oh.
Me (and my VIM) were absolutely unaware of this, so I've very likely fudged regular spaces over some of them.

I've now edited my vimrc to show NBSPs.

**Comment by pjotrek-b** [comment]
I can go through the document again and use NBSPs accordingly.

**Comment by JeromeMartinez** [comment]
@michaelni any formatting preference about NBSP, i.e. should we keep them?
On my side I prefer removing them for simplicity but not against having them.
then we stop with this space formatting stuff :).

**Comment by pjotrek-b** [comment]
btw: I just saw that in not all pseudo-code blocks in the initial ffv1.md use NBSPs, but regular space.

And sorry for opening this can of worms, but who could have known? :cry:

**Comment by michaelni** [comment]
> @michaelni any formatting preference about NBSP, i.e. should we keep them?

No oppinion :)

**Comment by pjotrek-b** [comment]
I think plain spaces will probably suffice in this case.
Since the width of the pseudo-code blocks is limited to <80 characters anyways, I expect it to be less likely to be broken in different renderings.

I'll replace it with spaces for the time being.

**Comment by pjotrek-b** [comment]
Why does noone tell me about the whitespace landmines set across this document, before you let me walk in, looking like a bull in a china shop? :cry: :wink:

Therefore I kept is as separate commit.
Can you cherry pick?

**Comment by michaelni** [comment]
I see commits with identical and duplicated commit messages (537b947d05dfe461451ed69059c6c0d513c098af, a9c64342a0e04448bcfa16c79624e727fcaf7aeb ). That obviously proofs that these commit messages are insufficiently describing the change, because they work for more than one change.
also i can cherry pick approved changes but it may be quicker and avoid mistakes (on my side) if pull requests do not need to be edited. Also if cherry picking is wanted refering to a commit by its hash is more foolproof than describing it
Anyway ill cherry pick them in order upto the first where i see an issue

**Comment by michaelni** [comment]
622bab9ecb013f265a343c286e8b7230015cbb1b adds spaces around if ( get_bits(1) ) { 
2b9c6fab0a8962f53c271b48403f14fab6a5e62d removes the spaces around if ( get_bits(1) ) { 

Making a change in a commit that is undone in the same pull request in a later commit is ugly and makes the changes a mess and hard to follow

**Comment by pjotrek-b** [comment]
I hope I got it fixed now? :sweat_smile: 

I've rebased it, reworded duplicate commit messages and removed "undo" commits by applying the final change in the first commit already (then skipped in rebase).

**Comment by retokromer** [comment]
@pjotrek-b I guess there is still [`get_sr_golomb(k)`](https://github.com/FFmpeg/FFV1/blob/master/ffv1.md#signed-golomb-rice-codes) to be unified.

**Comment by pjotrek-b** [comment]
I'd leave that one as it is.

Normalizing that one seemed bloated to me.
It's clear to read, and every code has its justified 1-or-2-liner-ifs :wink:

**Comment by michaelni** [comment]
44b780418d276726246e569a2d396f07eb21db31 is a merge, this does not belong in a pull request

**Comment by pjotrek-b** [comment]
Sorry :(
Don't know how that merge happened.

I'm not very experienced at all with git pull requests or branching/merging...

I think the easiest would be for me to start over and leave this messed up pull request behind :wink:

**Comment by pjotrek-b** [comment]
I think/hope I've removed the merge now.

I'll try to rebase it to resolve the conflict without merging. Wish me luck! :sweat_smile:
(Thanks @JeromeMartinez)

**Comment by pjotrek-b** [comment]
Should be fine now.
Please check :smile:

**Comment by michaelni** [comment]
generated pdf is filled with gibberish. like ".# Abstract" or "category = "std" litterally in the document

**Comment by dericed** [comment]
@michaelni I've been unable to reproduce the gibberish. Also his PR only impacts ffv1.md and not the frontmatter files or the Makefile so I'm not sure how you're getting a gibberish pdf from this branch.

**Comment by michaelni** [comment]
@dericed, i cannot reproduce it either now. I assume this was PEBCAK on my side, ill take another look tomorrow and if i dont spot anything then ill apply

**Comment by dericed** [comment]
nudge :)
